### DISCUSSION TITLE
two on-premises clusters connectivity in clusterset

### DISCUSSION THREAD

**Discussion by jinglina** [submariner-io/submariner#1979] [General]
in a clusterset with on-premises and public clusters included, is it necessary to guarantee they are IP reachable (two on-premises clusters) with public or private ips?  If they are not ip reachable, is it possible to establish connections between clusters across broker cluster?

**Comment by jinglina** [comment]
For example, in clusterset, there are four clusters: broker-cluster, public cluster and two  on-premises(they are not ip reachable using public ip or private ip)。 How the two on-premises clusters can be conencted with each other? If ip reachable is necessary with public or private ip, how to solve this kind of cluster combination? is it necessary doing something using broker cluster ?

**Comment by sridhargaddam** [comment]
You can find various typologies and what is supported and what is not in the following link - https://submariner.io/operations/nat-traversal/

If the two onPrem clusters are not reachable using publicIP or privateIPs, then connections cannot be established between these two clusters. Currently Submariner uses Broker only for exchanging the metadata and not as a transit router, so we cannot do anything via the Broker.

**Reply by jinglina** [reply]
As this way, how to deal with the scenario with two-premise clusters in clusterset. This is strict conditions.

**Reply by sridhargaddam** [reply]
Since Broker is not used for datapath (i.e., like a transit router) and Submariner creates a full-mesh of connections between all the clusters in the ClusterSet, as per the current design, there should be either public or private IP connectivity for the connections to be established at least in one direction.
